crates/dojo-language-server/src/bin/language_server.rs

```diff
@@ -17,15 +18,22 @@ async fn main() {
     #[cfg(feature = "runtime-agnostic")]
     let (stdin, stdout) = (stdin.compat(), stdout.compat_write());
 
+    let metadata = read_metadata(None).unwrap_or_else(|error| {
+        panic!("Problem reading metadata: {error:?}");
+    });
+
     let mut plugins = get_default_plugins();
     plugins.push(Arc::new(DojoPlugin { world_config: WorldConfig::default() }));
     plugins.push(Arc::new(StarkNetPlugin {}));
 
-    let db = RootDatabase::builder().detect_corelib().with_plugins(plugins).build().unwrap_or_else(
-        |error| {
-            panic!("Problem creating language database: {error:?}");
-        },
-    );
+    let mut db =
```

Comment:
once we have the scarb compiler, i think we want the db building to happen there

Reply:
I'm not sure language server should ever touch scarb compiler, because LS should be able to handle code written for old Cairo versions. Otherwise, editor updates would break users experience until they upgrade their code.